lib/elixir/test/elixir/kernel/errors_test.exs

```diff
@@ -660,12 +660,10 @@ defmodule Kernel.ErrorsTest do
   end
 
   test "doc attributes format" do
-    message = "expected moduledoc attribute given in the {line, doc} format, got: \"Other\""
+    message = "expected @moduledoc to be a binary, a boolean, or nil, got: 'foo'"
     assert_raise ArgumentError, message, fn ->
       defmodule DocAttributesFormat do
-        @moduledoc "ModuleTest"
-        {666, "ModuleTest"} = Module.get_attribute(__MODULE__, :moduledoc)
```

Comment:
Also, we still need to check for the previous error message. The issue is that `@moduledoc` automatically inserts the line but if you call Module.put_attribute(...) directly, you need to give the line yourself. So I think we need to have two tests: one that checks we raise if we don't have a tuple and another one for binary. They will be raised at different times for different uses of the API.